### DISCUSSION TITLE
Linux,  Avalonia 11: Can't debug application, app freezes after start, debug output shows DBus-exception

### DISCUSSION THREAD

**Discussion by bmscodespace** [AvaloniaUI/Avalonia#16827] [Q&A]
Hi, 
I have the following setup:
Linux Debian Bookworm,
Avalonia 11.0.10
IDEs:
Rider IDE 2024.1.6 
or 
VS Code

The code in question runs without any problems when I am in non-debug mode. While debugging is no problem in Windows, when I try to run the app in debug mode on Linux, I am getting the following error in the debug console
```
[IME] Unable to create IME input context:
Tmds.DBus.Protocol.DBusException: org.freedesktop.DBus.Error.UnknownMethod: Method "CreateInputContext" with signature "s" on interface "org.freedesktop.IBus.Portal" doesn't exist
   at Tmds.DBus.Protocol.DBusConnection.MyValueTaskSource`1.System.Threading.Tasks.Sources.IValueTaskSource<T>.GetResult(Int16 token)
   at Tmds.DBus.Protocol.DBusConnection.CallMethodAsync[T](MessageBuffer message, MessageValueReader`1 valueReader, Object state)
   at Tmds.DBus.Protocol.Connection.CallMethodAsync[T](MessageBuffer message, MessageValueReader`1 reader, Object readerState)
   at Avalonia.FreeDesktop.DBusIme.IBus.IBusX11TextInputMethod.Connect(String name)
   at Avalonia.FreeDesktop.DBusIme.DBusTextInputMethodBase.OnNameChange(Exception e, ValueTuple`3 args) (IBusX11TextInputMethod #50223649)
```
and the app freezes, the fan goes up (this happens in both VS Code  and the Rider IDE). 
The view in question contains several TextBox-elements which act as filter inputs for a table whose rows are filtered according to the filters set. Everything is refreshed when the filter input changes. Old "filter tasks" are cancelled when a new one is set.
Besides some "Binding errors" that happen earlier in the debug log and the one mentioned above there are no other errors as it seems.
Could the error mentioned above cause the problem i am having?

Thank you for any answer ;)

ps ... I have tested the code just now also on Avalonia 11.1.3. The error above seems to have gone in the debug log, but the problem remains ;(

**Comment by guilhermeolisi** [comment]
Hello, I have the same problem in my project.  On Windows the project runs correctly, but on Ubuntu 20, the same problem occurs with the IME (Tmds.DBus.Protocol.DBusException). I'm using Avalonia 11.3

**Comment by thevortexcloud** [comment]
My guess is you are missing a portal/IME provider. Please make sure you have one installed.
